src/client.ts

```diff
@@ -35,6 +36,7 @@ export default class Client implements ClientInterface {
     | undefined;
 
   private terminal: vscode.Terminal | undefined;
+  private windows = false;
```

Comment:
Let's avoid this extra property and just check every time we restart the server.